articles/building-a-cicd-pipeline-using-gitlab/index.md

```diff
@@ -0,0 +1,117 @@
+
+### Building a CI/CD pipeline
+[Continuous Integration](https://codeship.com/continuous-integration-essentials) can be defined as a process, in the software development lifecycle, where engineers and developers push code into a git-based repository. If developers do this very frequently, an automated process to verify and build the code is essential to save time and costs. This automated process of building and testing the code whenever a new commit is made is called CI or Continuous Integration. According to the [official documentation](https://docs.gitlab.com/ee/ci/), GitLab CI/CD is a powerful tool built into GitLab that allows you to apply all the continuous methods (Continuous Integration, Delivery, and Deployment) to your software with no third-party application or integration needed. 
+
+Users can configure GitLab CI/CD by creating a file called [.gitlab-ci.yml](https://docs.gitlab.com/ee/ci/yaml/gitlab_ci_yaml.html) in the root directory of the project. This file is written in a simple and easy to use language called YAML. [This](https://www.tutorialspoint.com/yaml/index.htm) is a good article on YAML and its basics. Once you define the .gitlab-ci.yml file, Gitlab creates a [pipeline](https://docs.gitlab.com/ee/ci/pipelines/index.html), which runs whenever there is a change to the code in the repository. These Pipelines can have single or multiple stages that run one after the other(in series). Each stage can consist of multiple jobs that are executed in parallel by the [gitlab-runner](https://docs.gitlab.com/runner/), which is an application that works with GitLab to run jobs in a pipeline.
+
+### Code
+Let us build a simple CI pipeline to run a python script whenever we push changes to our repository. 
+
+Create a python script called `test.py` in the repository and add the following lines of code into the file. The script prints "hello world" and "testing" 5 times. This is just an example to help you understand how to create and run pipelines. You can run complex scripts and commands as well. 
+
+```python
+
+if __name__ == '__main__':
+  
+  print('hello world')
+  
+  for i in range(5):
+    print('testing', i)
+
+```
+
+Create a file called .gitlab-ci in the repository and add the following snippet of code. The [`variables`](https://docs.gitlab.com/ee/ci/variables/) section defines a list of variables that can be used in the pipeline. I have declared a variable called `example`, whose value is set to "this is an example variable".  Variables are declared as key-value pairs(separated by a semicolon) in YAML. 
+
+The [`stages`](https://docs.gitlab.com/ee/ci/yaml/#stage) section defines a list of stages in the pipeline. These stages are executed one after the other, and hence, it is important to order them based on dependencies. In other words, stages that are dependant on other stages must be declared after all of its dependencies. 
```

Comment:
instead of "after all of its dependencies" it should be "after all of their dependencies"